src/components/smart_objects/src/smart_object.cc

```diff
@@ -723,16 +723,28 @@ void SmartObject::duplicate(const SmartObject& OtherObject) {
 void SmartObject::cleanup_data() {
   switch (m_type) {
     case SmartType_String:
-      delete m_data.str_value;
+      if (m_data.str_value) {
```

Comment:
@AGaliuzov 
> 1. C++ standard allows NULL-pointer deletion.
> Please see C++98 standard 5.3.5 Delete paragraph 2.

I didn't mean that delete NULL-pointer is not allowed, I just mean if is already NULL, the program will no nothing, in case delete two times which is a disaster.
> 2. Could you please let me know in which way if-condition helps to prevent memory leak?

I mean this not a reason for ```if``` but for ```NULL```
Thanks!